frontend/src/components/custom-tools/export-tool/ExportTool.jsx

```diff
@@ -0,0 +1,228 @@
+import {
+  DeleteOutlined,
+  QuestionCircleOutlined,
+  UserOutlined,
+} from "@ant-design/icons";
+import {
+  Avatar,
+  List,
+  Modal,
+  Popconfirm,
+  Radio,
+  Select,
+  Typography,
+} from "antd";
+import PropTypes from "prop-types";
+import { useEffect, useState } from "react";
+import "./ExportTool.css";
+
+import { SpinnerLoader } from "../../widgets/spinner-loader/SpinnerLoader";
+
+const SHARE_ALL = "share_all";
+const SHARE_CUSTOM = "share_custom";
+
+function ExportTool({
+  open,
+  setOpen,
+  toolDetails,
+
+  loading,
+  allUsers,
+  onApply,
+}) {
+  const [selectedUsers, setSelectedUsers] = useState([]);
+  const [filteredUserList, setFilteredUserList] = useState([]);
+  const [sharingOption, setSharingOption] = useState(SHARE_ALL);
+
+  useEffect(() => {
+    if (toolDetails?.shared_users) {
+      // set the selectedUsers to the IDs of shared users
+      // filter owners and shared users
+
+      const promptStudioUsers = toolDetails?.prompt_studio_users.map((user) =>
+        user?.id?.toString()
+      );
+
+      const users = allUsers.filter((user) => {
+        const userId = user?.id?.toString();
+        return (
+          !selectedUsers.includes(userId) &&
+          !promptStudioUsers.includes(userId) &&
+          toolDetails?.created_by?.toString() !== userId
+        );
+      });
+
+      setFilteredUserList(users);
+      toolDetails?.shared_to_org
+        ? setSharingOption(SHARE_ALL)
+        : setSharingOption(SHARE_CUSTOM);
+    }
+  }, [toolDetails, allUsers, selectedUsers]);
+
+  useEffect(() => {
+    if (toolDetails?.shared_users) {
+      setSelectedUsers(
+        toolDetails.shared_users
+          .filter((user) => {
+            const promptStudioUsers = toolDetails?.prompt_studio_users.map(
+              (user) => user?.id?.toString()
+            );
+            const userId = user?.id?.toString();
+            return (
+              !promptStudioUsers.includes(userId) &&
+              toolDetails?.created_by?.toString() !== userId
+            );
+          })
+          .map((user) => user?.id?.toString())
+      );
+    }
+  }, [toolDetails, allUsers]);
+
+  const handleDeleteUser = (userId) => {
+    setSelectedUsers((prevSelectedUsers) =>
+      prevSelectedUsers.filter((user) => user !== userId)
+    );
+  };
+  const filterOption = (input, option) =>
+    (option?.label ?? "").toLowerCase().includes(input.toLowerCase());
+
+  const onChange = (e) => {
+    setSharingOption(e.target.value);
+  };
+
+  let sharedWithContent;
+  if (sharingOption === SHARE_ALL) {
+    sharedWithContent = <Typography.Text>Shared with everyone</Typography.Text>;
+  } else {
+    sharedWithContent = (
```

Comment:
@johnyrahul NIT: Can we handle this in a function like this? I think this approach would improve the readability. Feel free to ignore if you think otherwise.

```
const handleSharedContent = () => {

  if (sharingOption === SHARE_ALL) {
    return <Typography.Text>Shared with everyone</Typography.Text>;
  }

  return (
    // JSX Content
  )
}
```